### DISCUSSION TITLE
How many separate queries when using populate()

### DISCUSSION THREAD

**Discussion by vresetnikov** [Automattic/mongoose#12589] [Q&A] [answered]
Hello,

When using the mongoose's populate() function, how many separate queries are fired? 

I am asking this because having MongoDB on a separate instance I would like to cut the amount of roundtrips needed for the data to reduce the latency.

**Comment by vkarpov15** [comment, marked as answer]
1 per populated field by default. Unless you use cursors or `perDocumentLimit`. So, for example, `await User.find().populate('friends').populate('blogPosts')`, that executes 3 queries: 1 for users, 1 for friends, 1 for blog posts.
